### PR TITLE
Refactor FXIOS-8407 [v125] Replace linkButton in FakespotActionFooterView

### PR DESCRIPTION
**Pull request by bmihai23**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8407)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18634)

## :bulb: Description
- This PR simplifies the implementation of the `FakespotActionFooterView` class by refactoring the link button usage. 
- The changes include replacing the previous implementation with a more straightforward approach using the `LinkButton` class and a `LinkButtonViewModel` for configuration.

**Light Theme**
| Before Dropdown | After Dropdown | 
| ------------- | ------------- |
| Simulator Screenshot - iPhone 15 Pro - 2024-02-19 at 11 05 37 | Simulator Screenshot - iPhone 15 Pro - 2024-02-19 at 11 05 40 |

**Dark Theme**
| Before Dropdown | After Dropdown | 
| ------------- | ------------- |
| Simulator Screenshot - iPhone 15 Pro - 2024-02-19 at 11 07 19 | Simulator Screenshot - iPhone 15 Pro - 2024-02-19 at 11 07 21 |

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift [hunk 1]
```diff
@@ -95,9 +95,11 @@ public final class FakespotActionFooterView: UIView, ThemeApplicable {
...
 
     // MARK: - ThemeApplicable
-    public func applyTheme(theme: Common.Theme) {
-        linkButton.setTitleColor(theme.colors.actionPrimary, for: .normal)
-        linkButton.setTitleColor(theme.colors.actionPrimaryHover, for: .highlighted)
-        titleLabel.textColor = theme.colors.textSecondary
+    public func applyTheme(theme: Theme) {
+        let colors = theme.colors
+        linkButton.applyTheme(theme: theme)
+        linkButton.setTitleColor(colors.actionPrimary, for: .normal)
```
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift:101]: The title color doesn't have to be set anymore as the `LinkButton` has the correct styles already.

File: firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift [hunk 2]
```diff
@@ -32,7 +32,7 @@ public struct FakespotActionFooterViewModel {
 public final class FakespotActionFooterView: UIView, ThemeApplicable {
     private struct UX {
         static let labelSize: CGFloat = 13
-        static let buttonSize: CGFloat = 13
+        static let buttonSize: CGFloat = 16
```
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift:35]: The font size of the button should stay at 13.

**Review by thatswinnie — COMMENTED** [review]
Thanks for taking the task on @bmihai23 🙏 

Could you please take a look at the padding of the link? This is the state in the current release which doesn't match the images you posted.
IMG_0016

**Comment by bmihai23** [comment]
Hi @thatswinnie, I wanted to reach out and let you know that I've addressed all the modifications you suggested and I've updated the screenshots to reflect the latest changes accurately.

File: firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift [hunk 3]
```diff
@@ -66,11 +61,17 @@ public final class FakespotActionFooterView: UIView, ThemeApplicable {
...
 
+        titleLabel.text = viewModel.title
         titleLabel.accessibilityIdentifier = viewModel.a11yTitleIdentifier
-        linkButton.accessibilityIdentifier = viewModel.a11yActionIdentifier
+
+        let linkButtonViewModel = LinkButtonViewModel(
+            title: viewModel.actionTitle,
+            a11yIdentifier: viewModel.a11yActionIdentifier,
+            fontSize: UX.buttonSize,
+            contentHorizontalAlignment: .leading
```
[the remark this answers is not in the document]
Comment (bmihai23) [reply firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift]: Hi @thatswinnie, I've implemented the changes you suggested for the contentInsets in the UX struct. Thank you 🙏

**Review by thatswinnie — APPROVED** [review]

**Comment by thatswinnie** [comment]
@bmihai23 could you please update your branch with main? The build is failing because your branch is not up-to-date and there are changes on main that are required by the build system to build correctly.

**Pull request merged**